eZ/Publish/Core/Repository/TrashService.php

```diff
@@ -130,9 +131,18 @@ public function trash(Location $location)
             throw $e;
         }
 
-        return isset($spiTrashItem)
-            ? $this->buildDomainTrashItemObject($spiTrashItem)
-            : null;
+        // Use sudo as we want a trash item regardless of user access to the trash.
+        try {
+            return isset($spiTrashItem)
+                ? $this->repository->sudo(
+                    function () use ($spiTrashItem) {
+                        return $this->buildDomainTrashItemObject($spiTrashItem);
+                    }
+                )
+                : null;
+        } catch (Exception $e) {
+            return null;
+        }
```

Comment:
Whole section above: Using `sudo()` here at all is debatable. Is it violating sane access principles? As an alternative, we could instead add a 2nd, optional parameter, like `trash($location, $locationTrashed)` and pass it by reference. Then the [SignalSlot TrashService](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/SignalSlot/TrashService.php#L91) can tell the difference between "deleted permanently" and "trashed but editor does not have read access to trash". Other suggestions?

Reply:
Alternative: trash() could return true when the location was trashed but the editor doesn't have trash access. This seems more sensible permission-wise.